apis/cache/v1alpha1/cachecluster_types.go

```diff
@@ -0,0 +1,347 @@
+
+	// The URL of the web page where you can download the latest ElastiCache client
+	// library.
+	ClientDownloadLandingPage string `json:"clientDownloadLandingPage,omitempty"`
+
+	// Represents a Memcached cluster endpoint which, if Automatic Discovery is
+	// enabled on the cluster, can be used by an application to connect to any node
+	// in the cluster. The configuration endpoint will always have .cfg in it.
+	ConfigurationEndpoint Endpoint `json:"configurationEndpoint,omitempty"`
+
+	// Describes a notification topic and its status. Notification topics are used
+	// for publishing ElastiCache events to subscribers using Amazon Simple Notification
+	// Service (SNS).
+	NotificationConfiguration NotificationConfiguration `json:"notificationConfiguration,omitempty"`
+
+	// A group of settings that are applied to the cluster in the future, or that
+	// are currently being applied.
+	PendingModifiedValues PendingModifiedValues `json:"pendingModifiedValues,omitempty"`
+
+	// A flag that enables in-transit encryption when set to true.
+	TransitEncryptionEnabled bool `json:"transitEncryptionEnabled,omitempty"`
+}
+
+// CacheClusterParameters define the desired state of an AWS ElastiCache
+// Cache Cluster. Most fields map directly to an AWS ReplicationGroup:
+// https://docs.aws.amazon.com/AmazonElastiCache/latest/APIReference/API_CreateReplicationGroup.html#API_CreateReplicationGroup_RequestParameters
+type CacheClusterParameters struct {
+
+	// If true, this parameter causes the modifications in this request and any
+	// pending modifications to be applied, asynchronously and as soon as possible,
+	// regardless of the PreferredMaintenanceWindow setting for the cluster.
+	// If false, changes to the cluster are applied on the next maintenance reboot,
+	// or the next failure reboot, whichever occurs first.
+	// +optional
+	ApplyImmediately *bool `json:"applyImmediately,omitempty"`
+
+	// Specifies whether the nodes in this Memcached cluster are created in a single
+	// Availability Zone or created across multiple Availability Zones in the cluster's
+	// region.
+	// This parameter is only supported for Memcached clusters.
+	// +optional
+	AZMode *string `json:"azMode,omitempty"`
+
+	// The password used to access a password protected server.
+	// +optional
+	AuthToken *string `json:"authToken,omitempty"`
+
+	// Specifies the strategy to use to update the AUTH token. This parameter must
+	// be specified with the auth-token parameter. Possible values:
+	// +optional
+	AuthTokenUpdateStrategy *string `json:"authTokenUpdateStrategy,omitempty"`
+
+	// The node group (shard) identifier. This parameter is stored as a lowercase
+	// string.
+	// CacheClusterID is a required field
+	// +optional
+	CacheClusterID string `json:"cacheClusterId"`
+
+	// A list of cache node IDs to be removed.
+	// +optional
+	CacheNodeIdsToRemove []string `json:"cacheNodeIdsToRemove,omitempty"`
+
+	// The compute and memory capacity of the nodes in the node group (shard).
```

Comment:
```suggestion
	// The compute and memory capacity of the nodes in the node group (shard).
	// +immutable
```

Reply:
`CacheNodeType` can actually be modified.